lab_2/main.py

```diff
@@ -89,7 +171,34 @@ def predict_language_knn(unknown_text_vector: list, known_text_vectors: list,
     :param k: the number of neighbors to choose label from
     :param metric: specific metric to use while calculating distance
     """
-    pass
+    if (not isinstance(unknown_text_vector, list)
+            or not isinstance(known_text_vectors, list)
+            or not isinstance(language_labels, list)):
+        return None
+    if len(language_labels) != len(known_text_vectors):
+        return None
+    for number in unknown_text_vector:
+        if not isinstance(number, (float, int)):
+            return None
+    distances = []
+    if metric == 'manhattan':
+        for known_text_vector in known_text_vectors:
+            distances.append(calculate_distance_manhattan(unknown_text_vector, known_text_vector))
+    if metric == 'euclid':
```

Comment:
If one variable is compared in several `if`s, it is usually done with `elif` - no need to create another `if` block - use the existing one.